crates/types/src/blockchain/header.rs

```diff
@@ -112,26 +112,31 @@ impl BlockHeader {
     /// Set the block height for the header
     pub fn set_block_height(&mut self, height: BlockHeight) {
         self.consensus_mut().height = height;
+        self.recalculate_metadata();
     }
 
     /// Set the previous root for the header
     pub fn set_previous_root(&mut self, root: Bytes32) {
         self.consensus_mut().prev_root = root;
+        self.recalculate_metadata();
```

Comment:
Should we use a builder in the hot paths instead of calling these setters one by one, now that they all have to recalculate the application header hash? We could save several hashing calls in the executor by calculating this after all the fields are set.

Reply:
We use these methods only in the tests, so it doesn't hurt us=) I will put them under `test-helper` feature